### PR TITLE
List rights for a given role.

### PR DESCRIPTION
**Pull request by sahithi**
Command sample output:
> vcd role list-rights 'vApp User'
name
---------------------------------
vApp Template: Checkout
UI Plugins: View
vApp: Snapshot Operations
vApp: Copy
vApp: Manage VM Password Settings
vApp Template / Media: View
vApp: Edit VM Properties
vApp: View VM metrics
vApp: Delete
vApp: Use Console
Disk: View Properties
vApp: Power Operations
vApp: Sharing
vApp: Edit Properties
vApp: Edit VM Network

### CODE REVIEW

File: pyvcloud/vcd/role.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        """  # NOQA
+        if self.resource is None:
+            self.resource = self.client.get_resource(self.href)
+        rights = []
+        if hasattr(self.resource, 'RightReferences') and \
+                hasattr(self.resource.RightReferences, 'RightReference'):
+            for right in self.resource.RightReferences.RightReference:
+                rights.append({'name': right.get('name')})
+            return rights
+        raise Exception("No rights found for the role \'%s\'" % self.name)
```
Comment (anusuyar) [inline pyvcloud/vcd/role.py]: Should we throw an exception when there are no rights in the role?(is it possible to have role with no rights?)  I think it's better to return the rights even if it is empty.

File: pyvcloud/vcd/role.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        self.client = client
+        self.href = href
+        self.resource = resource
+        if resource is not None:
+            self.href = resource.get('href')
+            self.name = resource.get('name')
+
+    def list_rights(self):
+        """
+        List rights of the role
```
Comment (rocknes) [inline pyvcloud/vcd/role.py]: nit : List rights associated with the role

File: pyvcloud/vcd/role.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        :return: list of names of rights for a given role.
```
Comment (rocknes) [inline pyvcloud/vcd/role.py]: Shouldn't we also send back the id of the right?

File: pyvcloud/vcd/role.py (hunk 1 above)
Comment (rocknes) [reply pyvcloud/vcd/role.py]: ^ +1.

File: pyvcloud/vcd/role.py (hunk 2 above)
Comment (sahithi) [reply pyvcloud/vcd/role.py]: done

File: pyvcloud/vcd/role.py (hunk 1 above)
Comment (sahithi) [reply pyvcloud/vcd/role.py]: Valid point. Modified the code.

File: pyvcloud/vcd/role.py (hunk 3 above)
Comment (sahithi) [reply pyvcloud/vcd/role.py]: User can retrieve ids using search command. 
I quickly checked all other commands with list sub-command, none of them is listing ids of any resources.

**Comment by sahithi** [comment]
Addressed comments

File: pyvcloud/vcd/role.py (hunk 2 above)
Comment (sahithi) [reply pyvcloud/vcd/role.py]: done

File: pyvcloud/vcd/role.py (hunk 3 above)
Comment (sahithi) [reply pyvcloud/vcd/role.py]: None of the existing list subcommands' (org, vapp etc) lists ids.
Ids can always be retrieved through search command.

File: pyvcloud/vcd/role.py (hunk 1 above)
Comment (sahithi) [reply pyvcloud/vcd/role.py]: Fixed it.

**Pull request merged**